Framework/Algorithms/src/FindPeaksConvolve.cpp

```diff
@@ -0,0 +1,376 @@
+  int startAdj{0};
+
+  if (sliceStart < 0) {
+    startAdj = -sliceStart;
+    adjPeakExtentBinNumber = peakExtentBinNumber - startAdj;
+    sliceStart = 0;
+  }
+  if (sliceStart + adjPeakExtentBinNumber > static_cast<size_t>(yData.size()) - 1) {
+    adjPeakExtentBinNumber = static_cast<size_t>(yData.size()) - sliceStart;
+  }
+
+  Eigen::VectorXd::Index maxIndex;
+  if (m_findHighestDatapointInPeak) {
+    const auto unweightedYData = EigenMap_const(yData.data() + sliceStart, adjPeakExtentBinNumber);
+    unweightedYData.maxCoeff(&maxIndex);
+  } else {
+    Eigen::VectorXd pdf{generateNormalPDF(static_cast<int>(peakExtentBinNumber))};
+    const auto weightedYData = EigenMap_const(yData.data() + sliceStart, adjPeakExtentBinNumber)
+                                   .cwiseProduct(EigenMap_const(pdf.data() + startAdj, adjPeakExtentBinNumber));
+    weightedYData.maxCoeff(&maxIndex);
+  }
+  return static_cast<size_t>(maxIndex) + sliceStart;
+}
+
+Eigen::VectorXd FindPeaksConvolve::generateNormalPDF(const int peakExtentBinNumber) {
+  // assert vector has odd size.
+  Eigen::VectorXd pdf{peakExtentBinNumber};
+  boost::math::normal_distribution<> dist(0.0, peakExtentBinNumber / 2.0); // assures 2 stddevs in the resultant vector
+  const int meanIdx{peakExtentBinNumber / 2};
+  for (int i{0}; i < peakExtentBinNumber; ++i) {
+    int x{i - meanIdx};
+    pdf(i) = boost::math::pdf(dist, x);
+  }
+  return pdf;
+}
+
+void FindPeaksConvolve::createIntermediateWorkspaces(const size_t dataIndex, const Tensor1D &kernel,
+                                                     const Tensor1D &iOverSigma, const EigenMap_const &xData) {
+  API::Algorithm_sptr alg{createChildAlgorithm("CreateWorkspace")};
+
+  alg->setProperty("OutputWorkspace", "iOverSigma");
+  alg->setProperty("DataX", std::vector<double>(xData.data(), xData.data() + xData.rows()));
+  alg->setProperty("DataY", std::vector<double>(iOverSigma.data(), iOverSigma.data() + iOverSigma.size()));
+  alg->execute();
+  API::MatrixWorkspace_sptr algOutput = alg->getProperty("OutputWorkspace");
+  API::AnalysisDataService::Instance().add("iOverSigma" + std::to_string(dataIndex), algOutput);
+
+  std::vector<double> xKernelData(kernel.size());
+  std::iota(std::begin(xKernelData), std::end(xKernelData), 0.0);
+  alg->resetProperties();
+  alg->setProperty("OutputWorkspace", "kernelWorkspace");
+  alg->setProperty("DataX", std::move(xKernelData));
+  alg->setProperty("DataY", std::vector<double>(kernel.data(), kernel.data() + kernel.size()));
+  alg->execute();
+  API::MatrixWorkspace_sptr algKernelOutput = alg->getProperty("OutputWorkspace");
+  API::AnalysisDataService::Instance().add("kernelWorkspace" + std::to_string(dataIndex), algKernelOutput);
+}
+
+void FindPeaksConvolve::outputResults() {
+  API::ITableWorkspace_sptr table{API::WorkspaceFactory::Instance().createTable("TableWorkspace")};
+  table->addColumn("int", "SpecIndex");
+  for (size_t i{0}; i < m_maxPeakCount; i++) {
```

Comment:
I think this is confusing as there is no easy way to tell how many peaks you've just found and it's quite difficult to read off the peak centres in this table
![image](https://github.com/mantidproject/mantid/assets/55979119/016f65f8-0b87-41fd-8e90-4db4e5533ec6)
We could have a table each for centre, height and I/sigma?

Reply:
Also this peak height appears to be the y-value found at the peak position - not minus any background! So I'm not sure how useful this would if e.g. you wanted to use it as an initial parameter in a fit!

Reply:
Ah I forgot about the background, shall I rename this peakYValue - do you think it's worth trying to calculate the peak height?

Reply:
I think you could probably extract it form the integrated intensity (from the y value convolution) and then assume it is a Gaussian etc. to convert from area to height using a sigma taken whether from the zero crossing (which as discussed offline is 2 sigma between inflection points) , or from the kernel size. But I wonder what it's useful for?

Reply:
I tend to use validation to guard against exceptions, and use warnings to advise if making math/logic errors like this. I'm happy to change to an error if this is preferred though. Would probably have to be set to 1.5*kernelsize-2.

Reply:
If that's OK I think it should be checked in the validation. Although it may not crash it could lead to large hangs (e.g. if user confuses units and gives an extent in TOF when the unit of the workspace is wavelength) and in the end it will only produce nonsense. The algorithm itself is only valid if the kernel size does not exceed the data extent. I would be happy to limit it to `kernelsize - 1` - the invalid region of the second (smaller smoothing kernel) convolution should only affect the invlaid region of the first convolution/